catroid/src/main/java/org/catrobat/catroid/embroidery/DSTFileGenerator.java

```diff
@@ -0,0 +1,244 @@
+			0x269, 0x268, 0x26a, 0x215, 0x214, 0x216, 0x211, 0x210, 0x212, 0x219, 0x218, 0x21a, 0x205, 0x204, 0x206,
+			0x201, 0x200, 0x202, 0x209, 0x208, 0x20a, 0x225, 0x224, 0x226, 0x221, 0x220, 0x222, 0x229, 0x228, 0x22a,
+			0x295, 0x294, 0x296, 0x291, 0x290, 0x292, 0x299, 0x298, 0x29a, 0x285, 0x284, 0x286, 0x281, 0x280, 0x282,
+			0x289, 0x288, 0x28a, 0x2a5, 0x2a4, 0x2a6, 0x2a1, 0x2a0, 0x2a2, 0x2a9, 0x2a8, 0x2aa};
+
+	private float minX;
+	private float maxX;
+	private float minY;
+	private float maxY;
+	private FileOutputStream fileStream;
+
+	ArrayList<PointF> stitchPoints = null;
+
+	public DSTFileGenerator() {
+		stitchPoints = new ArrayList<>();
+	}
+
+	public File createDSTFile(ArrayList<PointF> stitchPoints) {
+		if (stitchPoints.size() > 1) {
+			String filename = ProjectManager.getInstance().getCurrentProject().getName() + ".dst";
+			File dstFile = new File(Constants.CACHE_DIR, filename);
+
+			if (dstFile.exists()) {
+				dstFile.delete();
+			}
+			try {
+				dstFile.createNewFile();
+			} catch (IOException e) {
+				Log.e(TAG, Log.getStackTraceString(e));
+			}
+
+			prepareFileWriting(stitchPoints);
+			writeToDSTFile(dstFile);
+			return dstFile;
+		}
+		return null;
+	}
+
+	private void prepareFileWriting(ArrayList<PointF> stitchPoints) {
+		this.stitchPoints = convertStitchPointsUnit(stitchPoints);
+		calculateBoundingBox();
+	}
+
+	private void writeToDSTFile(File dstFile) {
+		try {
+			fileStream = new FileOutputStream(dstFile);
+
+			writeHeader();
+			writeStitches();
+			writeEndFile();
+
+			fileStream.flush();
+			fileStream.close();
+		} catch (IllegalArgumentException ex) {
+			Log.e(TAG, "Remaining stitch distance should be zero!");
+		} catch (IOException e) {
+			Log.e(TAG, Log.getStackTraceString(e));
+		}
+	}
+
+	private ArrayList<PointF> convertStitchPointsUnit(ArrayList<PointF> stitchPoints) {
+		ArrayList<PointF> convertedStitchPoints = new ArrayList<>();
+
+		for (PointF stitch: stitchPoints) {
+			PointF stitchUnitMm = new PointF();
+			stitchUnitMm.x = Math.round(stitch.x * STITCHPOINTUNITFACTOR);
+			stitchUnitMm.y = Math.round(stitch.y * STITCHPOINTUNITFACTOR);
+			convertedStitchPoints.add(stitchUnitMm);
+		}
+		return convertedStitchPoints;
+	}
+
+	private void calculateBoundingBox() {
+		initBoundingBox(stitchPoints.get(0));
+		for (PointF point: stitchPoints) {
+			if (minX > point.x) {
+				minX = point.x;
+			}
+			if (maxX < point.x) {
+				maxX = point.x;
+			}
+			if (minY > point.y) {
+				minY = point.y;
+			}
+			if (maxY < point.y) {
+				maxY = point.y;
+			}
+		}
+	}
+
+	private void initBoundingBox(PointF point) {
+		minX = point.x;
+		maxX = point.x;
+		minY = point.y;
+		maxY = point.y;
+	}
+
+	private void writeHeader() throws IOException {
+		String header = createHeaderString();
+
+		String headerFill = "";
+		for (int index = header.getBytes().length; index < HEADERMAXBYTE; index++) {
```

Comment:
you could use something like

`repeated = new String(new char[n]).replace("\0", s);`

from [stackoverflow](https://stackoverflow.com/questions/1235179/simple-way-to-repeat-a-string-in-java) (obviously). :sweat_smile:

Reply:
Guava ftw! [Strings.repeat(String string, int count)](https://google.github.io/guava/releases/23.0/api/docs/com/google/common/base/Strings.html#repeat-java.lang.String-int-)